src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -408,8 +408,11 @@ internal void UpdateCallables(IEnumerable<QsCallable> updates)
                     var specializations = this.GlobalSymbols.DefinedSpecializations(header.QualifiedName).Select(defined =>
                     {
                         var specHeader = defined.Item2;
-                        var compiledSpecs = compiled.Specializations.Where(spec => spec.Kind == specHeader.Kind);
-                        QsCompilerError.Verify(compiledSpecs.Count() <= 1, "more than one specialization of the same kind exists"); // currently not supported
+                        var compiledSpecs = compiled.Specializations.Where(spec => 
+                            spec.Kind == specHeader.Kind &&
+                            spec.TypeArguments.Equals(specHeader.TypeArguments)
```

Comment:
Is `spec.TypeArguments.Equals(specHeader.TypeArguments)` necessary for handling redeclared specializations or is this part of something else?

Reply:
We could potentially have a specializations for particular type arguments. The symbol table is set up to in fact also handle that a "bundle" of specializations (i.e. all specializations for particular type arguments) may have different characteristics. We pretty much want to treat the operation specializied for particular type arguments as if if was it's own declaration.